src/NServiceBus.Core/OpenTelemetry/Metrics/Meters.cs

```diff
@@ -16,4 +16,10 @@ class Meters
 
     internal static readonly Counter<long> TotalFailures =
         NServiceBusMeter.CreateCounter<long>("nservicebus.messaging.failures", description: "Total number of messages processed unsuccessfully by the endpoint.");
+
+    internal static readonly Histogram<double> ProcessingTime =
+     NServiceBusMeter.CreateHistogram<double>("nservicebus.messaging.processingtime", "ms", "The time in milliseconds between when the message was pulled from the queue until processed by the endpoint.");
+
+    internal static readonly Histogram<double> CriticalTime =
+        NServiceBusMeter.CreateHistogram<double>("nservicebus.messaging.criticaltime", "ms", "The time in milliseconds between when the message was sent until processed by the endpoint.");
```

Comment:
Any chance this will make it into the NSB v9 release that is RTM?

Reply:
@bbrandt, we started working on some of those metrics in https://github.com/Particular/NServiceBus/pull/7048. The goal is to release them in `9.1`. We don't have an ETA yet, but we're actively working on it.